### ISSUE TITLE
overloads in `if TYPE_CHECKING` shouldn't require implementations

### ISSUE THREAD

**Issue by A5rocks** [python/mypy#19169]
**Feature**

As I explained in the title.

**Pitch**

Currently this raises an error:

```py
from typing import TYPE_CHECKING, overload

if TYPE_CHECKING:
    @overload  # E: An overloaded function outside a stub file must have an implementation
    def f(x: int) -> str:
        ...
    
    @overload
    def f(x: str) -> bytes:
        ...
```

I think that if code is in `if TYPE_CHECKING` then it shouldn't need an implementation though.
